code/modules/vehicles/mecha/equipment/tools/work_tools.dm

```diff
@@ -72,66 +76,63 @@
 		clamptarget.set_anchored(TRUE)
 		if(!do_after_cooldown(target, source))
-			clamptarget.set_anchored(initial(clamptarget.anchored))
+			clamptarget.set_anchored(FALSE)
 			return
-		clamptarget.forceMove(workmech.cargo_hold)
 		clamptarget.set_anchored(FALSE)
+		clamptarget.forceMove(workmech.cargo_hold)
 		if(!chassis.ore_box && istype(clamptarget, /obj/structure/ore_box))
 			chassis.ore_box = clamptarget
 		to_chat(source, "[icon2html(src, source)][span_notice("[target] successfully loaded.")]")
 		log_message("Loaded [clamptarget]. Cargo compartment capacity: [workmech.cargo_hold.cargo_capacity - workmech.cargo_hold.contents.len]", LOG_MECHA)
+		return ..()
 
-	else if(isliving(target))
-		var/mob/living/M = target
-		if(M.stat == DEAD)
-			return
+	if(!isliving(target))
+		return ..()
 
-		if(!source.combat_mode)
-			step_away(M,chassis)
-			if(killer_clamp)
-				target.visible_message(span_danger("[chassis] tosses [target] like a piece of paper!"), \
-					span_userdanger("[chassis] tosses you like a piece of paper!"))
-			else
-				to_chat(source, "[icon2html(src, source)][span_notice("You push [target] out of the way.")]")
-				chassis.visible_message(span_notice("[chassis] pushes [target] out of the way."), \
-				span_notice("[chassis] pushes you aside."))
-			return ..()
-		else if(LAZYACCESS(modifiers, RIGHT_CLICK) && iscarbon(M))//meme clamp here
-			if(!killer_clamp)
-				to_chat(source, span_notice("You longingly wish to tear [M]'s arms off."))
-				return
-			var/mob/living/carbon/C = target
-			var/torn_off = FALSE
-			var/obj/item/bodypart/affected = C.get_bodypart(BODY_ZONE_L_ARM)
-			if(affected != null)
-				affected.dismember(damtype)
-				torn_off = TRUE
-			affected = C.get_bodypart(BODY_ZONE_R_ARM)
-			if(affected != null)
-				affected.dismember(damtype)
-				torn_off = TRUE
-			if(!torn_off)
-				to_chat(source, span_notice("[M]'s arms are already torn off, you must find a challenger worthy of the kill clamp!"))
-				return
+	var/mob/living/victim = target
+	if(victim.stat == DEAD)
+		return
+
+	if(!source.combat_mode)
+		step_away(victim, chassis)
+		if(killer_clamp)
+			target.visible_message(span_danger("[chassis] tosses [target] like a piece of paper!"), \
+				span_userdanger("[chassis] tosses you like a piece of paper!"))
+		else
+			to_chat(source, "[icon2html(src, source)][span_notice("You push [target] out of the way.")]")
+			chassis.visible_message(span_notice("[chassis] pushes [target] out of the way."), \
+			span_notice("[chassis] pushes you aside."))
+		return ..()
+
+	if(iscarbon(victim))//meme clamp here
+		var/mob/living/carbon/carbon_victim = target
+		var/torn_off = FALSE
+		var/obj/item/bodypart/affected = carbon_victim.get_bodypart(BODY_ZONE_L_ARM)
+		if(affected != null)
+			affected.dismember(damtype)
+			torn_off = TRUE
+		affected = carbon_victim.get_bodypart(BODY_ZONE_R_ARM)
+		if(affected != null)
+			affected.dismember(damtype)
+			torn_off = TRUE
```

Comment:
```suggestion
		var/obj/item/bodypart/affected = carbon_victim.get_bodypart(BODY_ZONE_L_ARM) || carbon_victim.get_bodypart(BODY_ZONE_R_ARM)
		if(!isnull(affected))
			affected.dismember(damtype)
			torn_off = TRUE
```
Does this work? I think it'd be nicer than how it is currently written

Reply:
For bonus points make it not always rip the left arm first 😄

Reply:
actually doing this means I think you don't even need `var/torn_off` any more

Reply:
disregard im stupid

Reply:
happens to best of us